pkg/kapp/resources/file_resources.go

```diff
@@ -94,3 +106,38 @@ func (r FileResource) Resources() ([]Resource, error) {
 
 	return resources, nil
 }
+
+// isDir returns if path is a directory. If fsys is nil, isDir calls os.Stat(path); otherwise, it checks path inside
+// fsys.
+func isDir(fsys fs.FS, path string) (bool, error) {
+	if fsys == nil {
+		fileInfo, err := os.Stat(path)
+		if err != nil {
+			return false, fmt.Errorf("error stat'ing file %q: %v", path, err)
```

Comment:
**nit**: Since the error from os.Stat() is of the format `stat <file-name>: error`, we could directly return err instead.
(Same in other places in isDir)